### DISCUSSION TITLE
Multiple Tailwind CSS classes having multiple Webpack entry points

### DISCUSSION THREAD

**Discussion by sharukhkhanajm** [tailwindlabs/tailwindcss#4751] [Help]
please read the problem statement carefully in order to answer the question.

**Problem statement**

So I have a React project setup with webpack and tailwind CSS.  
In my webpack config I have multiple entry point in order to generate different CSS and JS for each entry point.  
The problem arises when I use the tailwind classes in my React components.  
Let's suppose if I use a tailwind class  **bg-red-600**  only in Component1(or entry point 1).  
So after building the files through webpack the  **bg-red-600**  will be present in all the entry point's generated CSS files(keep in mind I have just used this class in first entry point component only).  
What it should be doing is only have  **bg-red-600**  class in first component CSS file instead it is preset in all the CSS files even though I have not used it in any other place other than first component.

Hope I was able to made my point.

Looking forward to your replies. Thanks.

Webpack's entry points:

```
entry: {
    app1: path.resolve(
        __dirname,
        'src/Component1'
    ),
    app2: path.resolve(
        __dirname,
        'src/Component2'
    ),

},
```

**Comment by myang5** [comment]
What does your `tailwind.config.js` file look like? If your config is set up to purge all files in both `app1` and `app2`, classes from both entry points will end up together in the same CSS file
